File: app/lib/features/kyc/domain/entities/kyc.dart
/// The four KYC verification states a customer can be in. Drives which status
/// card the `/kyc` screen renders and the [StatusBadge] colour.
enum KycStatus { unverified, pending, verified, rejected }

/// The document types a customer can submit for verification.
enum KycDocumentType { passport, idCard, license }

/// Wire value for a document type (matches the API's `documentType`).
extension KycDocumentTypeWire on KycDocumentType {
  String get apiValue => switch (this) {
        KycDocumentType.passport => 'passport',
        KycDocumentType.idCard => 'id_card',
        KycDocumentType.license => 'license',
      };
}

/// Parses the API's `documentType` string into a [KycDocumentType] (defaults to
/// [KycDocumentType.idCard]).
KycDocumentType kycDocumentTypeFromWire(String? value) => switch (value) {
      'passport' => KycDocumentType.passport,
      'license' => KycDocumentType.license,
      _ => KycDocumentType.idCard,
    };

/// Parses the API's `status` string into a [KycStatus] (defaults to unverified).
KycStatus kycStatusFromWire(String? value) => switch (value) {
      'pending' => KycStatus.pending,
      'verified' => KycStatus.verified,
      'approved' => KycStatus.verified,
      'rejected' => KycStatus.rejected,
      _ => KycStatus.unverified,
    };

/// The form payload submitted for verification — personal background info plus
/// the uploaded document-photo URLs (front always; back required for
/// id_card/license, optional for passports).
class KycSubmission {
  const KycSubmission({
    required this.fullName,
    required this.dateOfBirth,
    required this.placeOfBirth,
    required this.placeOfResidence,
    required this.documentType,
    required this.documentNumber,
    required this.documentFrontUrl,
    this.documentBackUrl,
  });

  final String fullName;

  /// ISO `yyyy-mm-dd`.
  final String dateOfBirth;
  final String placeOfBirth;
  final String placeOfResidence;
  final KycDocumentType documentType;
  final String documentNumber;

  /// Public URL returned by `POST /kyc/documents` for the front photo.
  final String documentFrontUrl;

  /// Back-photo URL; null only when the document is a passport.
  final String? documentBackUrl;
}

/// The stored details of a customer's KYC submission, echoed back by
/// `GET /kyc/me` so the profile can display what was submitted. Null until the
/// customer has filed a submission.
class KycSubmissionDetails {
  const KycSubmissionDetails({
    required this.fullName,
    required this.dateOfBirth,
    required this.placeOfBirth,
    required this.placeOfResidence,
    required this.documentType,
    required this.documentNumber,
    this.documentFrontUrl,
    this.documentBackUrl,
  });

  final String fullName;

  /// ISO `yyyy-mm-dd`.
  final String dateOfBirth;
  final String placeOfBirth;
  final String placeOfResidence;
  final KycDocumentType documentType;
  final String documentNumber;
  final String? documentFrontUrl;
  final String? documentBackUrl;
}

/// The customer's derived KYC profile. [rejectionReason] is present only when
/// [status] is [KycStatus.rejected]; [submission] carries the submitted details
/// once the customer has filed one (used to display them on the profile).
class KycProfile {
  const KycProfile({
    required this.status,
    this.rejectionReason,
    this.submission,
  });

  final KycStatus status;
  final String? rejectionReason;
  final KycSubmissionDetails? submission;
}

File: admin/src/features/kyc/api/kyc.ts
import { apiClient } from '@/lib/api-client'
import type { ApiResponse, PaginationMeta } from '@/types'

/** Moderation state of a KYC submission. */
export type KycStatus = 'pending' | 'approved' | 'rejected'

/** Identity-document type. */
export type KycDocumentType = 'passport' | 'id_card' | 'license'

/** A KYC submission as returned by the admin API (the record + account contact). */
export interface AdminKyc {
  id: string
  userId: string
  userContact: string
  fullName: string
  dateOfBirth: string
  placeOfBirth: string
  placeOfResidence: string
  documentType: KycDocumentType
  documentNumber: string
  documentFrontUrl?: string
  documentBackUrl?: string
  status: KycStatus
  rejectionReason?: string
  reviewedBy?: string
  reviewedAt?: string
  createdAt: string
  updatedAt: string
}

export interface KycListParams {
  page?: number
  limit?: number
  status?: KycStatus | ''
  q?: string
}

const ADMIN = '/api/admin/kyc'

export function listKyc(
  params: KycListParams,
): Promise<ApiResponse<AdminKyc[]> & { meta?: PaginationMeta }> {
  const q = new URLSearchParams()
  if (params.page !== undefined) q.set('page', String(params.page))
  if (params.limit !== undefined) q.set('limit', String(params.limit))
  if (params.status) q.set('status', params.status)
  if (params.q) q.set('q', params.q)
  const qs = q.toString()
  return apiClient.get<AdminKyc[]>(qs ? `${ADMIN}?${qs}` : ADMIN)
}

export function setKycStatus(
  id: string,
  status: KycStatus,
  rejectionReason = '',
): Promise<ApiResponse<AdminKyc>> {
  return apiClient.put<AdminKyc>(`${ADMIN}/${id}`, { status, rejectionReason })
}
